pkg/controller/daemon/daemon_controller.go

```
@@ -519,6 +526,63 @@ func (dsc *DaemonSetsController) updatePod(old, cur interface{}) {
 	}
 }
 
+// listSuspendedDaemonPods lists the Daemon pods that 'want to run, but should not schedule'
+// for the node.
+func (dsc *DaemonSetsController) listSuspendedDaemonPods(node string) (dss []string) {
+	dsc.suspendedDaemonPodsLock.Lock()
+	defer dsc.suspendedDaemonPodsLock.Unlock()
+
+	if _, found := dsc.suspendedDaemonPods[node]; !found {
+		return []string{}
+	}
+
+	for k := range dsc.suspendedDaemonPods[node] {
+		dss = append(dss, k)
+	}
+	return
+}
+
+// requeueSuspendedDaemonPods enqueues all DaemonSets which has pods that 'want to run,
+// but should not schedule' for the node; so DaemonSetController will sync up them again.
+func (dsc *DaemonSetsController) requeueSuspendedDaemonPods(node string) {
+	dss := dsc.listSuspendedDaemonPods(node)
+	for _, dsKey := range dss {
+		if ns, name, err := cache.SplitMetaNamespaceKey(dsKey); err != nil {
+			continue
+		} else if ds, err := dsc.dsLister.DaemonSets(ns).Get(name); err == nil {
```

when will there be an error? it is not good to simply ignore error in general.